### PR TITLE
Stop using the cgi module, deprecated in Python 3.11 per PEP 594

### PR DESCRIPTION
**Pull request by scop**
## What do these changes do?

Removes use of the `cgi` module which is deprecated in 3.11+ per https://peps.python.org/pep-0594/#cgi

## Are there changes in behavior for the user?

No

## Related issue number

No

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by scop** [comment]
Untested and mostly for discussion purposes for now, quite likely some refinements would be in order.

File: aiohttp/helpers.py [hunk 1]
```diff
@@ -748,7 +748,11 @@ def _parse_content_type(self, raw: str) -> None:
             self._content_type = "application/octet-stream"
             self._content_dict = {}
         else:
-            self._content_type, self._content_dict = cgi.parse_header(raw)
+            msg = HeaderParser().parsestr("Content-Type: " + raw)
+            self._content_type = msg.get_content_type()
```
Comment (scop) [inline aiohttp/helpers.py:752]: One difference compared to cgi is that if the actual type is invalid (for example `foo`), this returns `text/plain`.

Comment (Dreamsorcerer) [reply aiohttp/helpers.py:752]: Documentation here suggest to use `Message`: https://peps.python.org/pep-0594/#cgi
Though the result looks much the same.

File: aiohttp/helpers.py [hunk 2]
```diff
@@ -748,7 +748,11 @@ def _parse_content_type(self, raw: str) -> None:
...
+            params = msg.get_params()
+            del params[0]  # content type duplicated here
+            self._content_dict = dict(params)
```
Comment (Dreamsorcerer) [inline aiohttp/helpers.py]:
```suggestion
            self._content_dict = dict(params[1:])  # First element is content type again
```
Maybe a list slice is simpler?

File: aiohttp/helpers.py (hunk 1 above)
Comment (scop) [reply aiohttp/helpers.py:752]: Yeah, I know.

But to me the _example_ (it's labeled as such, not a suggestion) in the PEP isn't that great -- to parse a HTTP header one would create an e-mail Message, set a header to it and then grab the header's parameters back... quite ugly in my opinion. The approach I took here to me reads much better: parse a header using a header parser.

Another reason using a Message doesn't sound that great to me in new code is that it is labeled as a compat/legacy thing:
- https://docs.python.org/3/library/email.compat32-message.html#compat32-message
- https://docs.python.org/3/library/email.message.html#id2

File: aiohttp/helpers.py (hunk 2 above)
Comment (scop) [reply aiohttp/helpers.py]: Yes, definitely, thanks!

**Comment by Dreamsorcerer** [comment]
If you could also handle the backport, that would be great.

**Comment by scop** [comment]
https://github.com/aio-libs/aiohttp/pull/6734

**Comment by webknjaz** [comment]
Backported to 3.8 as https://github.com/aio-libs/aiohttp/commit/cb79f0dc1597b683ab5a5a9fed8262954a086d35

**Pull request merged**